.github/workflows/release.yaml

```diff
@@ -0,0 +1,38 @@
+name: Release
```

Comment:
@lizan are you okay with this approach or prefer the release push to be part of https://github.com/envoyproxy/gateway/blob/main/.github/workflows/build_and_test.yaml to reduce dup ?

Reply:
This is fine. Though we don't need to run all build steps, instead of that, just use some image tool (e.g. `skopeo` to copy `envoyproxy/gateway-dev:<git_sha>` to `envoyproxy/gateway:<git_tag>`, and upload manifests generated by #300 to github release.

Reply:
I prefer if we modify the existing workflow file and add another section similar to `push` in https://github.com/envoyproxy/gateway/blob/b77b8789518027308474898defa4227e562800d5/.github/workflows/build_and_test.yaml#L45 rather than pushing to the registry and pulling again (using an intermediate tool like `skopeo`) on a different runner.

Reply:
It's not pushing to the registry and pull again, the build/push happens when a commit merged (which is the current build_and_test). A tag workflow should just do the tag.

Reply:
Yes, the release workflow is triggered by tag creating not commit merged. It makes sense to separate it with build_and_test workflow.

Reply:
One more thing needs to confirm, do we need to keep make lint and make go unit test in releasing workflows? @lizan

Reply:
I see the lint and unit test targets in the PR. I'm fine with keeping them.

Reply:
its a dup with build_and_test.yaml, but should be okay in short term and can be deduped later

Reply:
Ok, let me remove this.